scripts/devtools/tpl-manager.py

```diff
@@ -288,21 +305,21 @@ def concretize_spec(self, check_spec):
             if (self.args.add_spec):
                 add_cmd = SpackCommand("add")
                 add_cmd(self.args.spec)
-        conc_cmd = SpackCommand("concretize")
-        conc_args = []
+        force_conc = False
         if (self.args.clean):
-            conc_args.append("-f")
+            force_conc = True
             print("Cleaning and concretizing environment")
         else:
             print("Concretizing environment")
-        conc_cmd(*conc_args)
+        conc_spec = self.spack_env.concretize(force=force_conc)
         if (check_spec):
             matches = self.spack_env.matching_spec(self.spack_spec)
             if (not matches):
                 raise Exception(f"{self.args.spec} not found in current "+\
                                 "environment. Rerun with --add-spec to add it.")
             self.spack_spec = matches
             print(f"Found matching root spec for {self.args.spec}")
+        self.spack_env.write()
```

Comment:
Why does this need to be done after the concretization completes and why wasn't it needed before?